saleor/payment/migrations/tasks/saleor3_12.py

```diff
@@ -310,3 +311,27 @@ def set_default_currency_for_transaction_event_task():
         TransactionItem, TransactionEvent, BATCH_SIZE
     ):
         set_default_currency_for_transaction_event_task.delay()
+
+
+def update_transaction_token_field(transaction_item_class, batch_size=None):
+    transactions = transaction_item_class.objects.filter(token__isnull=True)
+    fully_processed = True
+    if batch_size:
+        transactions = transactions[:BATCH_SIZE]
+
+    for transaction in transactions:
+        transaction.token = RandomUUID()
```

Comment:
You can do this in a single update query, take a look [here](https://github.com/saleor/saleor/blob/main/saleor/checkout/migrations/0044_fulfill_checkout_line_new_fields.py#L15)

Reply:
Right, thanks

Reply:
@IKarbowiak Hhm, I am not sure if this the best way for handling this. By calling `.update` and `Case` I will lock the whole table with `transactionItem`. And I don't think that I can use `update` with slice `[:BATCH_SIZE]`. I believe that the celery worker should be fast, take part of the records, update them and move to the next batch.
Am I wrong?